service/debugger/locations.go

```diff
@@ -0,0 +1,291 @@
+package debugger
+
+import (
+	"debug/gosym"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/derekparker/delve/service/api"
+)
+
+const maxFindLocationCandidates = 5
+
+type LocationSpec interface {
+	Find(d *Debugger, locStr string) ([]api.Location, error)
```

Comment:
The implementation of this method should always return fully filled out Location structs. I noticed some only return Addr filled out.